easybuild/easyblocks/b/boost.py

```diff
@@ -181,18 +191,16 @@ def build_step(self):
 
         if self.cfg['boost_mpi']:
             self.log.info("Building boost_mpi library")
+            self.build_boost_variant(bjamoptions + " --user-config=user-config.jam --with-mpi",paracmd)
+        
+        if self.cfg['boost_multi_thread']:
+            self.log.info("Building boost with multi threading")
+            self.build_boost_variant(bjamoptions + " threading=multi --layout=tagged",paracmd)
 
-            bjammpioptions = "%s --user-config=user-config.jam --with-mpi" % bjamoptions
-
-            # build mpi lib first
-            # let bjam know about the user-config.jam file we created in the configure step
-            run_cmd("./bjam %s %s" % (bjammpioptions, paracmd), log_all=True, simple=True)
-
-            # boost.mpi was built, let's 'install' it now
-            run_cmd("./bjam %s  install %s" % (bjammpioptions, paracmd), log_all=True, simple=True)
-            
-            # cleanup previous build before proceeding with the full Boost
-            run_cmd("./bjam --clean-all", log_all=True, simple=True)
+        # if both boost_mpi and boost_multi_thread are enabled, build boost mpi with multi-thread support
+        if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
+            self.log.info("Building boost_mpi with multi threading")
+            self.build_boost_variant(bjamoptions + " --user-config=user-config.jam --with-mpi threading=multi --layout=tagged",paracmd)
```

Comment:
@mboisson lines should be kept under 120 chars, so:

```python
if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
    self.log.info("Building boost_mpi with multi threading")
    extra_bjamoptions = " --user-config=user-config.jam --with-mpi threading=multi --layout=tagged"
    self.build_boost_variant(bjam_options + extra_bjamoptions, paracmd)
```


also, style nitpicking: space after `,` (before `paracmd`)